src/XPurchase/VirtoCommerce.XPurchase/CartValidationError.cs

```diff
@@ -20,6 +20,8 @@ public CartValidationError(IEntity entity, string error, string errorCode = null
         public string ObjectType { get; set; }
         public string ObjectId { get; set; }
         public List<ErrorParameter> ErrorParameters =>
-            FormattedMessagePlaceholderValues.Select(kvp => new ErrorParameter {Key = kvp.Key, Value = kvp.Value.ToString()}).ToList();
+            FormattedMessagePlaceholderValues
+                ?.Select(kvp => new ErrorParameter { Key = kvp.Key, Value = kvp.Value.ToString() })
+                ?.ToList();
```

Comment:
Redundant null check after `Select()`.
```suggestion
                .ToList();
```